common/src/main/java/io/druid/common/utils/SerializerUtils.java

```diff
@@ -39,29 +39,65 @@
   private static final Charset UTF8 = Charset.forName("UTF-8");
 
   /**
-   * Writes the given long value into the given OutputStream, using the helperBuffer. Faster alternative to
-   * out.write(Longs.toByteArray(value)), more convenient (sometimes) than wrapping the OutputStream into {@link
-   * java.io.DataOutputStream}. The order in which the value's bytes are written to the OutputStream depends on the
-   * {@link ByteBuffer#order(ByteOrder)} of the given buffer.
+   * Writes the given long value into the given OutputStream in big-endian byte order, using the helperBuffer. Faster
+   * alternative to out.write(Longs.toByteArray(value)), more convenient (sometimes) than wrapping the OutputStream into
+   * {@link java.io.DataOutputStream}.
    *
-   * @param helperBuffer a heap ByteBuffer of capacity at least 8.
+   * @param helperBuffer a big-endian heap ByteBuffer with capacity of at least 8
    */
-  public static void writeLongToOutputStream(OutputStream out, long value, ByteBuffer helperBuffer) throws IOException
+  public static void writeBigEndianLongToOutputStream(OutputStream out, long value, ByteBuffer helperBuffer)
+      throws IOException
   {
+    if (helperBuffer.order() != ByteOrder.BIG_ENDIAN) {
+      throw new IllegalArgumentException("Expected big-endian byteBuffer");
+    }
+    helperBuffer.putLong(0, value);
+    out.write(helperBuffer.array(), 0, 8);
```

Comment:
Might be nice to also null-check the `helperBuffer.array()` just to ensure that you actually get it.  As is, an NPE here could be because 

1. `out` is null
2. `helperBuffer` is null
3. `helperBuffer` is the wrong object type and therefore `.array()` is null